addons/mail/static/src/models/composer_view.js

```diff
@@ -561,7 +558,7 @@ registerModel({
                     return;
                 }
             }
-            if (this.messaging.currentPartner) { // not supported for guest
+            if (composer.thread.channel) {
```

Comment:
this shows typing is not declarative enough. Here we should guess typing applies only for channel, hence the conditional